standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -2863,6 +2862,10 @@ private boolean drop_table_core(final RawStore ms, final String catName, final S
         deletePartitionData(partPaths, ifPurge, ReplChangeManager.shouldEnableCm(db, tbl));
         // Delete the data in the table
         deleteTableData(tblPath, ifPurge, ReplChangeManager.shouldEnableCm(db, tbl));
+      } else if (TxnUtils.isTransactionalTable(tbl)) {
+        CompactionRequest rqst = new CompactionRequest(dbname, name, CompactionType.MAJOR);
```

Comment:
This is here to prepare for the Cleaner right?
I think it would be enough to put a record in "ready for cleaning" state
And you would need to pass to location I guess.
Plus you don't need to to this if the data was purged.

Reply:
Yeah, that's just a placeholder for marking table as "ready for cleaning".